backend/lib/edgehog/containers/manual_actions/update_status.ex

```diff
@@ -0,0 +1,138 @@
+#
+# This file is part of Edgehog.
+#
+# Copyright 2024 SECO Mind Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+defmodule Edgehog.Containers.ManualActions.UpdateStatus do
+  @moduledoc false
+  use Ash.Resource.ManualUpdate
+
+  alias Edgehog.Devices
+
+  @impl Ash.Resource.ManualUpdate
+  def update(changeset, _opts, context) do
+    changeset = check_images(changeset, context)
+    Ash.update(changeset)
```

Comment:
I don't like the current approach: reading these lines, it only looks like you're checking the images.
I would rather see all actions explicitly top level:

```suggestion
    changeset
    |> check_images(context)
    |> check_networks(context)
    |> check_containers(context)
    |> check_deployments(context)
    |> Ash.update()
```

Being changesets, it's easy to apply the next action only if the one before has succeeded. I see two possible ways to approach this:
- by using `Ash.Changeset.fetch_argument_or_change` to get the current status of the changeset (which could have been updated by previous actions), and only going forward with the check if the status is the expected status
- adding an error to the changeset in case not all resources are available. this allows removing all intermediate states (pulling_images, creating_networks, ...) if we want to go in that direction, but changes the semantic meaning of a failed `UpdateStatus.update`